src/Arcus.Messaging.Pumps.ServiceBus/Configuration/AzureServiceBusMessagePumpSettings.cs

```diff
@@ -105,32 +113,184 @@ private static ServiceBusEntityType ConvertToServiceBusEntityType(ServiceBusEnti
+        internal async Task<ServiceBusProcessor> CreateMessageProcessorAsync()
+        {
+            if (_tokenCredential is null)
+            {
+                string rawConnectionString = await GetConnectionStringAsync();
+                ServiceBusConnectionStringProperties serviceBusConnectionString = ServiceBusConnectionStringProperties.Parse(rawConnectionString);
+
+                var client = new ServiceBusClient(rawConnectionString);
+                {
+                    ServiceBusProcessor processor;
+                    if (string.IsNullOrWhiteSpace(serviceBusConnectionString.EntityPath))
+                    {
+                        // Connection string doesn't include the entity so we're using the message pump settings
+                        if (string.IsNullOrWhiteSpace(EntityName))
+                        {
+                            throw new ArgumentException("No entity name was specified while the connection string is scoped to the namespace");
+                        }
+
+                        processor = CreateProcessor(client, EntityName, SubscriptionName);
+                    }
+                    else
+                    {
+                        // Connection string includes the entity so we're using that instead of the message pump settings
+                        processor = CreateProcessor(client, serviceBusConnectionString.EntityPath, SubscriptionName);
+                    }
+
+                    return processor;
+                }
+            }
+            else
+            {
+                var client = new ServiceBusClient(FullyQualifiedNamespace, _tokenCredential);
+                ServiceBusProcessor processor = CreateProcessor(client, EntityName, SubscriptionName);
+
+                return processor;
+            }
+        }
+
+        private ServiceBusProcessor CreateProcessor(ServiceBusClient client, string entityName, string subscriptionName)
+        {
+            ServiceBusProcessorOptions options = DetermineMessageProcessorOptions();
+            
+            if (string.IsNullOrWhiteSpace(subscriptionName))
+            {
+                return client.CreateProcessor(entityName, options);
+            }
+
+            return client.CreateProcessor(entityName, subscriptionName, options);
+        }
+        
+        private ServiceBusProcessorOptions DetermineMessageProcessorOptions()
+        {
+            var messageHandlerOptions = new ServiceBusProcessorOptions();
+            if (Options != null)
+            {
+                // Assign the configured defaults
+                messageHandlerOptions.AutoCompleteMessages = Options.AutoComplete;
+                messageHandlerOptions.MaxConcurrentCalls = Options.MaxConcurrentCalls ?? messageHandlerOptions.MaxConcurrentCalls;
+            }
+
+            return messageHandlerOptions;
+        }
+        
+        private async Task<string> GetConnectionStringAsync()
```

Comment:
Thought, should we use these traces?
https://github.com/arcus-azure/arcus.messaging/pull/178/files#diff-21920603b21d912cc5678a977a854f57051464f5f3eff50f3dccfcd4d50abca5L454-L456

Reply:
Yeah, probably.